pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -41,6 +41,14 @@ type ProxierHealthUpdater interface {
 
 	// Run starts the healthz HTTP server and blocks until it exits.
 	Run() error
+
+	proxierHealthChecker
+}
+
+type proxierHealthChecker interface {
```

Comment:
This is not something you need to act on, but I wanted to call out a small distinction.

You can define this interface here, next to the "producer", and it's fine (it's all the same pkg, after all).  Or you can define it next to the consumer, embracing the Go idea of "produce types, accept interfaces".  This file has no use of this interface, but the service_health (and test) file gets simpler because of it.

The receiver is saying "I need something I can ask IsHealthy()", the producer is passing something that satisfies it.

Again, it's fine this way, too.

Reply:
I wasn't aware of these notions. You got me reading up on something :) I updated the PR with this